Make server URL configurable at build time

jouvent opened the pull request:
- add scons CPPDEFINE reading SERVICE_DOMAIN env variable
- define is as  constant in base/constants.h
- use it const_addresses.h for the various URL endpoints
- adapt URL endpoints for crystalnix/omaha-server

jouvent committed:
CPPDEFINE for update server url

* add scons CPPDEFINE reading SERVICE_DOMAIN env variable
* define is as  constant in base/constants.h
* use it const_addresses.h for the various url endpoints
* addapt url endpoints for crystalnix/omaha-server

aduewel commented on omaha/main.scons:
```diff
@@ -70,7 +70,9 @@ _SHORT_COMPANY_NAME = 'OmahaCompanyName'
 _PRODUCT_NAME = 'Update'
 _COMPANY_DOMAIN_BASE = 'google'
 _COMPANY_DOMAIN = _COMPANY_DOMAIN_BASE + '.com'
+_SERVICE_DOMAIN = os.getenv('SERVICE_DOMAIN', "update.googleapis.com")
```
the default value does not make much sense, since you change the path in `omaha/base/const_addresses.h` for `kUrlCrashReport`

perhaps its a better idea to have all the endpoints in `omaha/base/const_addresses.h`  as env variables (including the path segment) and fall back to the google default if they are not set

aduewel commented:
We should mention these new env-variables in [CustomizingOmaha](https://github.com/dentalwings/omaha-client/blob/master/doc/CustomizingOmaha.md), too.

jouvent committed:
URLs as env variables

- make every URLs in const_addresses initialized from a constant in 
constants.h
- URLs can optionaly be defined from env var in scons script
